laletterbuilder/tests/test_letter_sending.py

```diff
@@ -0,0 +1,57 @@
+from django.utils import timezone
+import pytest
+
+from users.tests.factories import UserFactory
+from .factories import HabitabilityLetterFactory, LandlordDetailsFactory
+from laletterbuilder.letter_sending import (
+    email_letter_to_landlord,
+    send_letter_via_lob,
+    create_letter,
+)
+from laletterbuilder.models import HabitabilityLetter
+
+
+def test_nothing_is_emailed_on_demo_deployment(settings, mailoutbox, db):
+    settings.IS_DEMO_DEPLOYMENT = True
+    letter = HabitabilityLetterFactory()
+    LandlordDetailsFactory(user=letter.user, email="landlordo@calrissian.net")
+    assert email_letter_to_landlord(letter, b"blah") is True
+    assert letter.letter_emailed_at is not None
+    assert len(mailoutbox) == 0
+
+
+def test_nothing_is_emailed_if_already_emailed(settings, mailoutbox):
+    settings.IS_DEMO_DEPLOYMENT = False
+    letter = HabitabilityLetter(letter_emailed_at=timezone.now())
+    assert email_letter_to_landlord(letter, b"blah") is False
+    assert len(mailoutbox) == 0
+
+
+def test_nothing_is_mailed_if_already_sent(settings):
+    settings.IS_DEMO_DEPLOYMENT = False
+    letter = HabitabilityLetter(letter_sent_at=timezone.now())
+    assert (
+        send_letter_via_lob(
+            letter,
+            b"blah",
+            letter_description="laletterbuilder letter",
+        )
+        is False
+    )
```

Comment:
These are great. Once we add the user's preference for mailing here, we should add some tests for that. e.g. if the user provides a landlord email, but says they do NOT want it emailed, make sure it's not emailed. And same for landlord address and physical mail.